src/Compilers/CSharp/Test/EndToEnd/EndToEndTests.cs

```diff
@@ -327,24 +327,25 @@ static void runTest(int n)
         public void LongInitializerList()
         {
             CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+            int iterations = 0;
             try
             {
-                initializerTest(cts.Token);
+                initializerTest(cts.Token, ref iterations);
             }
-            catch (TaskCanceledException)
+            catch (Exception e) when (e is OperationCanceledException or TaskCanceledException)
             {
-                Assert.True(false, "Test timed out while getting all semantic info for long initializer list");
+                Assert.True(false, $"Test timed out while getting all semantic info for long initializer list. Got to {iterations} iterations.");
```

Comment:
Why not `RoslynDebug.AssertOrFailFast` here? That way you get a heap dump when this happens.

Reply:
Because I didn't think of it.